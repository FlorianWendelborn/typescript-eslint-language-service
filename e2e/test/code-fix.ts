import path from "path";
import assert from "assert";
import { createServer, TSServer } from "../helper/server";

function findResponse(responses: any[], commandName: string) {
  return responses.find(response => response.command === commandName);
}

describe("LanguageService plugin", () => {
  let server: TSServer;

  afterEach(async () => {
    if (server) await server.close();
  });
  describe("#getCodeFixes", () => {
    it("should not return ESLint error when the project does not use @typescript-eslint/parser", async () => {
      server = createServer({ projectPath: path.resolve(__dirname, "../projects/simple") });
      const { file, fileContent } = server.readFile("./main.ts");
      server.send({ command: "open", arguments: { file, fileContent, scriptKindName: "TS" } });
      await server.waitEvent("projectLoadingFinish");
      server.send({ command:"getCodeFixes",arguments:{ file, startLine:1, startOffset:12, endLine:1, endOffset:12, errorCodes:[30010] } });
      await server.waitResponse("getCodeFixes");
      const found = findResponse(server.responses, "getCodeFixes");
      if (!found) {
        throw new assert.AssertionError();
      }
      expect(found).toMatchSnapshot();
    });
  });
});
