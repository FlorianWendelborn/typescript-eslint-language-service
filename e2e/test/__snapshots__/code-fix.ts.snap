// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageService plugin #getCodeFixes should not return ESLint error when the project does not use @typescript-eslint/parser 1`] = `
Object {
  "body": Array [
    Object {
      "changes": Array [
        Object {
          "fileName": "/Users/yosuke/workspace/javascript/typescript-eslint-language-service/e2e/projects/simple/main.ts",
          "textChanges": Array [
            Object {
              "end": Object {
                "line": 1,
                "offset": 12,
              },
              "newText": ";",
              "start": Object {
                "line": 1,
                "offset": 12,
              },
            },
          ],
        },
      ],
      "description": "fix: Missing semicolon.",
      "fixId": "semi",
      "fixName": "semi",
    },
  ],
  "command": "getCodeFixes",
  "request_seq": 2,
  "seq": 0,
  "success": true,
  "type": "response",
}
`;
